include/openssl/ssl.h

```diff
@@ -250,6 +250,31 @@ typedef int (*tls_session_secret_cb_fn) (SSL *s, void *secret,
                                          STACK_OF(SSL_CIPHER) *peer_ciphers,
                                          const SSL_CIPHER **cipher, void *arg);
 
+/* Extension context codes */
+/* This extension is only allowed in TLS */
+#define SSL_EXT_TLS_ONLY                        0x0001
+/* This extension is only allowed in DTLS */
+#define SSL_EXT_DTLS_ONLY                       0x0002
+/* Some extensions may be allowed in DTLS but we don't implement them for it */
+#define SSL_EXT_TLS_IMPLEMENTATION_ONLY         0x0004
```

Comment:
It took me a while and some reading of the implementation to figure out what this meant when it first went in as an internal flag.  (Just based on the name, I thought it was a flag used internally to the implementation and was not expected from external callers.)  I don't know if SSL_EXT_ONLY_IMPLEMENTED_FOR_TLS is enough more clear to be worth the extra characters, though.